// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package serviceextensions.cookie_manager;

// Cookie operation types
enum CookieOperation {
  COOKIE_OPERATION_UNSPECIFIED = 0;
  SET = 1;
  DELETE = 2;
  OVERWRITE = 3;
}

// Cookie configuration based on Apache log format
message CookieConfig {
  string name = 1;
  string value = 2;
  string domain = 3;
  string path = 4 [default = "/"];
  int32 max_age = 5 [default = -1];  // -1 for session cookie, >0 for persistent
  bool secure = 6;
  bool http_only = 7 [default = true];
  bool same_site_strict = 8;
  CookieOperation operation = 9 [default = SET];
}

// Configuration message containing multiple cookie configurations
message CookieManagerConfig {
  repeated CookieConfig cookies = 1;
}